transitions/index.js

```diff
@@ -262,127 +310,57 @@ var attach = function() {
-
-                /* Fetch entire changed document */
-                db.medic.get(id, function(e, doc) {
-                    if (e) {
-                        failures++;
-                        logger.error('backlog: fetch failed for %s (%s)', id, e);
-                        return _.delay(next_fn, backlog_delay);
-                    }
-                    change.doc = doc;
-                    if (hasTransitionErrors(doc) || !hasTransitions(doc)) {
-                        logger.debug(
-                            'backlog: processing matched on %s seq %s',
-                            id, change.seq
-                        );
-                        applyTransitions({
-                            change: change,
-                            audit: audit,
-                            db: db
-                        });
-                    } else {
-                        logger.debug(
-                            'backlog: processing skipped on %s seq %s',
-                            id, change.seq
-                        );
-                    }
-                    /* Next */
-                    processed++;
-                    return _.delay(next_fn, backlog_delay);
-                });
-            },
-
-            /* Completion handler */
-            function () {
-                logger.info(
-                    'backlog: processing complete, %d failures', failures
-                );
+        logger.info('backlog: fetching changes feed, starting from %s', processedSeq);
+        var feed = new follow.Feed({
+            db: process.env.COUCH_URL,
+            since: processedSeq
+        });
+        feed.on('change', function(change) {
+            // skip uninteresting documents
+            if (change.deleted || change.id.match(/^_design\//)) {
+                return;
             }
-        );
+            changeQueue.push(change);
+        });
+        feed.follow();
     });
-};
-
-var hasTransitions = function(doc) {
-    return Object.keys(doc.transitions || {}).length > 0;
-};
 
-var hasTransitionErrors = function(doc) {
-    var transitions = doc.transitions || {};
-    for (var key in transitions) {
-        if (transitions.hasOwnProperty(key)) {
-            if (!transitions[key].ok) {
-                return true;
-            }
+    var processChange = function(change, callback) {
+        if (!change) {
+            return callback();
         }
-    }
+        var id = change.id;
+        db.medic.get(id, function(err, doc) {
+            if (err) {
+                logger.error('backlog: fetch failed for %s (%s)', id, err);
+                return callback();
+            }
+            if (match_types.indexOf(doc.type) === -1) {
+                // not a doc we know how to transition - ignore
+                return callback();
+            }
+            if (processed > 0 && (processed % PROGRESS_REPORT_INTERVAL) === 0) {
+                logger.info('backlog: %d items processed', processed);
```

Comment:
So this can be a little confusing, because at some point the backlog has been processed and we start processing the current changes.  We might even remove the concept of "backlog" because there's only one change processor.  It's starts at 0 and records as it goes.  I think you can really remove the idea of "backlog processor" and just have a change processor that starts at whatever is in `sentinel-meta-data` and states that "starting at update_seq x", and "processed x changes" every so often.  That might simplify things.